api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1758,15 +1787,19 @@ type AzurePlatformSpec struct {
 	// +immutable
 	VnetID string `json:"vnetID,omitempty"`
 
-	// SubnetName is the name of an existing subnet in VnetName / VnetID where Azure CCM will find an existing load
-	// balancer to be used for node egress.
+	// SubnetID is the subnet ID of an existing subnet where the load balancer for node egress will be created. This
+	// subnet is expected to be a subnet within the VnetID / VnetName. This subnet is expected to exist under the same
+	// subscription as SubscriptionID.
 	//
-	// Example: if your SubnetID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/<subnetName>,
-	//          your SubnetName is <subnetName>
+	// This resource is expected to exist in VnetResourceGroupName. If VnetResourceGroupName is not provided, HyperShift
+	// will assume it is in ResourceGroupName.
```

Comment:
Same comment as https://github.com/openshift/hypershift/pull/3945/files#r1591922974 but applying to this attribute

Reply:
This is OBE with the removal of VnetResourceGroupName